app/controllers/api/v7/works_controller.rb

```diff
@@ -138,6 +158,14 @@ def get_ids(params)
       collection = Work.where(publisher_id: publisher.id)
     elsif params[:contributor_id] && contributor = Contributor.where(pid: params[:contributor_id]).first
       collection = Work.joins(:contributions).where("contributions.contributor_id = ?", contributor.id)
+    elsif params[:id]
+      id_hash = get_id_hash(params[:id])
+      if id_hash.respond_to?("key")
```

Comment:
This is a DataCite use case, enabling `works?id=10.5555/1234` in addition to `works/10.5555/1234`. In contrast to `params[:ids]` this is more flexible in what it accepts as input via `get_id_hash`.

The `respond_to?("key")` is also in the older /app/controllers/api/v7/base_controller.rb. This is similar to `id_hash.is_a?(Hash)`.